static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.tsx

```diff
@@ -315,6 +317,33 @@ class SentryApplicationDetails extends DeprecatedAsyncView<Props, State> {
     return tokensToDisplay;
   };
 
+  rotateClientSecret = async () => {
+    try {
+      const rotateResponse = await this.api.requestPromise(
+        `/sentry-apps/${this.props.params.appSlug}/rotate-secret/`,
+        {
+          method: 'POST',
+        }
+      );
+      openModal(({Body, Header}) => (
+        <Fragment>
+          <Header>{t('Rotated Client Secret')}</Header>
+          <Body>
+            <Alert type="info" showIcon>
+              {t('This will be the only time your client secret is visible!')}
+            </Alert>
+            <p>
+              {t('Your client secret is:')}
+              <code>{rotateResponse.clientSecret}</code>
+            </p>
+          </Body>
+        </Fragment>
```

Comment:
I think i created a component such that the value is only shown once, would that fit your needs? You wouldn't have to worry about wording and everything because you could just pass in the token